test/gpdb/scripts/test_pitr_gpdb6.sh

```diff
@@ -0,0 +1,231 @@
+MIRROR2=${DATADIR}/dbfast_mirror2/demoDataDir1
+MIRROR3=${DATADIR}/dbfast_mirror3/demoDataDir2
+MASTER_PORT=6000
+PRIMARY1_PORT=6002
+PRIMARY2_PORT=6003
+PRIMARY3_PORT=6004
+
+echo "Filling the pgbackrest.conf configuration file" > /dev/null
+cat <<EOF > $PGBACKREST_TEST_DIR/pgbackrest.conf
+[seg-1]
+pg1-path=$MASTER
+pg1-port=$MASTER_PORT
+
+[seg0]
+pg1-path=$PRIMARY1
+pg1-port=$PRIMARY1_PORT
+
+[seg1]
+pg1-path=$PRIMARY2
+pg1-port=$PRIMARY2_PORT
+
+[seg2]
+pg1-path=$PRIMARY3
+pg1-port=$PRIMARY3_PORT
+
+[global]
+repo1-path=$PGBACKREST_TEST_DIR/$TEST_NAME
+log-path=$PGBACKREST_TEST_DIR/logs/$TEST_NAME
+start-fast=y
+fork=GPDB
+EOF
+
+echo "Initiallizing pgbackrest for GPDB" > /dev/null
+for i in -1 0 1 2
+do 
+    PGOPTIONS="-c gp_session_role=utility" $PGBACKREST_BIN/pgbackrest \
+    --config $PGBACKREST_TEST_DIR/pgbackrest.conf --stanza=seg$i stanza-create
+done
+
+echo "Configuring WAL archiving command" > /dev/null
+gpconfig -c archive_mode -v on
+
+gpconfig -c archive_command -v "'PGOPTIONS=\"-c gp_session_role=utility\" \
+$PGBACKREST_BIN/pgbackrest --config $PGBACKREST_TEST_DIR/pgbackrest.conf \
+--stanza=seg%c archive-push %p'" --skipvalidation
+
+gpstop -ar
+
+echo "pgbackrest health check" > /dev/null
+for i in -1 0 1 2
+do 
+	PGOPTIONS="-c gp_session_role=utility" $PGBACKREST_BIN/pgbackrest \
+    --config $PGBACKREST_TEST_DIR/pgbackrest.conf --stanza=seg$i check
+done
+
+# The test scenario starts here
+echo "Creating initial dataset..." > /dev/null
+createdb gpdb_pitr_database
+psql -d gpdb_pitr_database -c \
+"CREATE TABLE t1 (id int, text varchar(255)) DISTRIBUTED BY (id);"
+
+psql -d gpdb_pitr_database -c \
+"INSERT INTO t1 SELECT i, 'text'||i FROM generate_series(1,30) i;"
+
+psql -d gpdb_pitr_database -c "SELECT * FROM t1 ORDER BY id;" \
+-o $PGBACKREST_TEST_DIR/$TEST_NAME/t1_rows_original.out
+
+echo "Creating full backup on master and seg0..." > /dev/null
+for i in -1 0
+do 
+    PGOPTIONS="-c gp_session_role=utility" $PGBACKREST_BIN/pgbackrest \
+    --config $PGBACKREST_TEST_DIR/pgbackrest.conf --stanza=seg$i \
+    --type=full backup
+done
+
+echo "Checking the presence of first backup" > /dev/null
+function check_backup(){
+    found=0
+    for dir in $PGBACKREST_TEST_DIR/$TEST_NAME/backup/seg$1/*; do
+        if [ -d "$dir" ]; then
+            dirname=$(basename "$dir")
+            current_date=$(date +%Y%m%d)
+            if [[ "$dirname" =~ ^${current_date}-[0-9]{6}F$ ]]; then
+                if [ "$(ls -A "$dir")" ]; then
+                    echo "Found a backup directory for segment $1: $dirname" \
+                    > /dev/null
+                    found=1
+                    break
+                fi
+            fi
+        fi
+    done
+    if [ $found -eq 0 ]; then
+        echo "The backup directory for segment $1 wasn't found" > /dev/null
+        exit 1
+    fi
+}
+for i in -1 0
+do 
+   check_backup $i
+done
+
+echo "Creating additional table..." > /dev/null
```

Comment:
Here is my vision. 
First backup:
we have created a backup for the data related to t1 only on seg0.
Second backup
we have created a backup for the data related to t2 only on seg1 and seg2
additionally we backup the rest of the data for t1 on those segments. And,potentially, the data on seg0 for t1 could have been changed by this time. 

And the main idea of the test is to check that after restoring the base backup we replay archived WAL logs for tables whose data wasn't backuped initially. 

For the table t2 we can see that on seg0 we have not created a backup, but PITR restored its data.
We could also show such behavior also for t1 probably, or do not backup t1 at all, but it will be meaningless backup

Reply:
But where do we get the WAL for t2 for segment 0?

Reply:
> And,potentially, the data on seg0 for t1 could have been changed by this time.

could you add it to test?

Reply:
I [added](https://github.com/arenadata/pgbackrest/blob/e819020882b54d9d07acd05571aed06636867f5a/test/gpdb/scripts/test_pitr_gpdb6.sh#L128-L135) the insertion of more data to t1 at seg0 after the first backup is done